backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/docvalues/impl/AbstractNumericDocValues.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.lowlevel.docvalues.impl;
+
+import org.apache.lucene.index.NumericDocValues;
+import org.apache.lucene.search.DocIdSetIterator;
+
+import java.io.IOException;
+
+/**
+ * Base implementation that throws an {@link IOException} for the
+ * {@link DocIdSetIterator} APIs. This impl is safe to use for sorting and
+ * aggregations, which only use {@link #advanceExact(int)} and
+ * {@link #longValue()}.
```

Comment:
Since this was copied from Elasticsearch (I think?), please give them credit, to avoid legal trouble:

```
 * <p>
 * Copied and adapted from {@code PUT THE ORIGINAL CLASS NAME HERE}
 * of <a href="https://github.com/elastic/elasticsearch">Elasticsearch project</a>.
```

Reply:
From what I see, a lot has been copied from there, practically the whole package is a copy. I corrected in the package where there was no such information.

Reply:
Corrected